py_src/ipyelk/elements/extended.py

```diff
@@ -75,43 +74,43 @@ def __getitem__(self, key):
             return edge
 
 
-@element
 class Record(Node):
-    layoutOptions: Dict = field(default_factory=lambda: {**record_opts})
-    width: float = field(default=80)
+    layoutOptions: Dict = Field(default_factory=lambda: {**record_opts})
+    width: float = Field(default=80)
 
-    def to_json(self):
+    def dict(self, **kwargs):
         # TODO need ability to resize the min width based on label/child max width
         for child in self.children:
             child.layoutOptions = merge(
                 opt.OptionsWidget(
                     options=[
                         opt.NodeSizeConstraints(),
-                        opt.NodeSizeMinimum(width=self.width, height=20),
+                        opt.NodeSizeMinimum(width=int(self.width), height=20),
```

Comment:
Maybe `width` should be `min_width` and `height` added as an attribute called `min_height`?

Reply:
Added a min_height but i think width is still cleaner. The intention is that this is width shared by all of the compartments. We might be able to set the width directly but then there is less feedback if the user "guessed" the wrong width (marks get rendered on top of each other instead of the compartment borders looking odd). There might be better ways to solve this in the future either with a dedicated "node:compartment" type (which has issues to being out of sync with how elkjs positioned the marks) or with multiple passes of layout calls to elkjs.